config/src/main/java/com/typesafe/config/impl/SerializedConfigValue.java

```diff
@@ -355,7 +380,12 @@ private static AbstractConfigValue readValueData(DataInput in, SimpleConfigOrigi
             String sd = in.readUTF();
             return new ConfigDouble(origin, vd, sd);
         case STRING:
-            return new ConfigString.Quoted(origin, in.readUTF());
+            int readTimes =  byteLength/ MAX_BYTES_LENGTH + ((byteLength % MAX_BYTES_LENGTH == 0) ? 0 : 1);
```

Comment:
writeUTF actually writes the 2 bytes of utf length before the string and this two bytes will be added to the byteLength, so it should be possible to construct the very long string value which will be divided to >32k strings during write and read attempt will throw `EOFException`.

One of the possible solution is to write (and read) a string chunks count before the first string.